src/cocoa/toga_cocoa/widgets/table.py

```diff
@@ -1,31 +1,58 @@
+from toga.sources import to_accessor
+
 from ..libs import *
 from .base import Widget
+from .utils import TogaData, TogaIconCell
 
 
 class TogaTable(NSTableView):
     # TableDataSource methods
     @objc_method
     def numberOfRowsInTableView_(self, table) -> int:
-        return len(self.interface.data.rows) if self.interface.data else 0
+        return len(self.interface.data) if self.interface.data else 0
 
     @objc_method
     def tableView_objectValueForTableColumn_row_(self, table, column, row: int):
-        column_index = int(column.identifier)
-        return self.interface.data.item(row, column_index)
+        row = self.interface.data[row]
+        try:
+            data = row._impls
+        except AttributeError:
+            data = {
+                attr: TogaData.alloc().init()
```

Comment:
Am I missing something, or is this dict comprehension creating blank `TogaData` objects and never filling in the data?

Reply:
Correct. The actual data is filled in when the row is rendered.